src/cargo/ops/cargo_rustc/job_queue.rs

```diff
@@ -209,12 +212,19 @@ impl<'a> JobQueue<'a> {
             // Any dirty stage which runs at least one command gets printed as
             // being a compiled package
             Dirty => {
-                self.printed.insert(pkg);
-                try!(config.shell().status("Compiling", pkg));
+                //check if documenting or compiling
+                if key.profile.doc {
+                    self.documented.insert(key.pkg);
+                    try!(config.shell().status("Documenting", key.pkg));
+                }
+                else {
```

Comment:
Stylistically we typically have the closing brace of the `if` and `else` on the same line